src/sentry/api/serializers/rest_framework/mentions.py

```diff
@@ -33,8 +38,8 @@ def separate_actors(actors):
     return {"users": users, "teams": teams}
 
 
-def separate_resolved_actors(actors):
-    users = [actor for actor in actors if isinstance(actor, User)]
+def separate_resolved_actors(actors: Sequence[APIUser | Team]):
+    users = [actor for actor in actors if isinstance(actor, BaseUser)]
```

Comment:
This could probably just be `APIUser` actually.  We only need `BaseUser` for codepaths that risk taking both ORMs and our dataclasses, but this function is utility and only comes from one callsite.